addons/hr_expense/models/hr_expense.py

```diff
@@ -101,8 +101,42 @@ def _get_employee_id_domain(self):
 
     is_editable = fields.Boolean("Is Editable By Current User", compute='_compute_is_editable')
     is_ref_editable = fields.Boolean("Reference Is Editable By Current User", compute='_compute_is_ref_editable')
-
+    is_nonzero_product_selected =  fields.Boolean("Is product with non zero cost selected", compute='_compute_is_nonzero_product_selected')
+    is_currency_id_different = fields.Boolean("Is currency_id different from the company_currency_id", compute='_compute_is_currency_id_different')
```

Comment:
I'd rename the fields (and adapt the name of the compute method), here are some suggestions:
 - `is_nonzero_product_selected`: `product_has_cost` (lack of imagination here)
 - `is_currency_id_different`: `same_currency`